src/libs/Request.ts

```diff
@@ -3,7 +3,7 @@ import enhanceParameters from './Network/enhanceParameters';
 import * as NetworkStore from './Network/NetworkStore';
 import Request from '../types/onyx/Request';
 
-type Middleware = (response: unknown, request: Request, isFromSequentialQueue: boolean) => Promise<unknown>;
+type Middleware = (response: Promise<unknown>, request: Request, isFromSequentialQueue: boolean) => Promise<unknown>;
```

Comment:
I switched this to be of type `Response` instead of `unknown`. It's a hopefully safe assumption but might not be true if we were to try to use `Request.processWithMiddleware` to interact with some random non-Expensify service.